core/types.mli

```diff
@@ -5,7 +5,8 @@ open CommonTypes
 type 'a stringmap = 'a Utility.StringMap.t [@@deriving show]
 type 'a field_env = 'a stringmap [@@deriving show]
 
-val print_types_pretty : bool Settings.setting
+type pretty_printer_engine = PPOld | PPRoundtrip | PPAll
```

Comment:
I don't think you need this, the `pretty_printer_engine` can be a private setting to the module. Thinking about it, I am not even convinced that `print_types_pretty` needs to be exposed. I think you can delete it as well.